tests/test_upload.py

```diff
@@ -58,6 +59,99 @@ def test_successful_upload(make_settings, capsys):
     assert captured.out.count(NEW_RELEASE_URL) == 1
 
 
+def test_successful_upload_add_gpg_signature(make_settings, capsys, monkeypatch):
+    """Test uploading package when gpg signature are added to it"""
+
+    # Create a custom package object and patch
+    # from_filename to return our custom patched package
+    package = package_file.PackageFile(
+        filename=helpers.WHEEL_FIXTURE,
+        comment=None,
+        metadata=pretend.stub(name="twine"),
+        python_version=None,
+        filetype=None,
+    )
+    monkeypatch.setattr(
+        package_file, "PackageFile", pretend.stub(from_filename=lambda *_: package),
+    )
+
+    # Stub create_repository to mock successful upload of package taking
+    # a pre-signed distribution
+    dists = [helpers.WHEEL_FIXTURE, helpers.WHEEL_FIXTURE + ".asc"]
```

Comment:
Nitpick: I had put this at the top of the test to give more context about what was being tested. I think it would also be fine to remove the `dists` variable and pass the list directly to `upload()`. I don't think it belongs here, where the response is being stubbed.

Reply:
I have put back `dists` to the top of the function in https://github.com/pypa/twine/pull/622/commits/458c42c2811559ba175f8d2a4bd60850ff59e75d